crates/trigger/src/lib.rs

```diff
@@ -369,6 +394,13 @@ impl<Executor: TriggerExecutor> TriggerAppEngine<Executor> {
             )
         })
     }
+
+    pub async fn resolve(
```

Comment:
Nit: maybe `resolve_template` here too, since it may not be obvious what an "engine" is "resolving".